harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/localfilesystem/LocalFsHarvesterFileVisitor.java

```diff
@@ -194,7 +194,7 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
                                 result.updatedMetadata++;
                             }
                         }
-                        idsForHarvestingResult.add(id);
+                        idsForHarvestingResult.add(new Integer(id));
```

Comment:
Use Integer.valueOf()

Using new Integer instead of Integer.valueOf() means a new integer object is created each time instead of using the Integer class. Always use the valueOf methods unless you have specific need to create a new object.